crates/pathfinder/src/core.rs

```diff
@@ -447,3 +452,46 @@ thin_starkhash_debug_display!(
     SequencerAddress,
     TransactionNonce,
 );
+
+#[cfg(test)]
+mod tests {
+    mod block_id_serde {
+        use super::super::BlockId;
+
+        #[derive(serde::Deserialize)]
+        struct Foo {
```

Comment:
Why do we need this wrapper struct for testing deserialization? I mean what's the value it adds vs just testing `BlockId`?